src/pudl/extract/ferc1.py

```diff
@@ -655,112 +493,20 @@ def get_raw_df(
 )
 def dbf2sqlite(context) -> None:
     """Clone the FERC Form 1 Visual FoxPro databases into SQLite."""
-    ferc1_to_sqlite_settings = (
-        context.resources.ferc_to_sqlite_settings.ferc1_dbf_to_sqlite_settings
-    )
-    datastore = context.resources.datastore
-    db_path = str(Path(context.op_config["pudl_output_path"]) / "ferc1.sqlite")
-    clobber = context.op_config["clobber"]
-
-    # Read in the structure of the DB, if it exists
-    logger.info("Dropping the old FERC Form 1 SQLite DB if it exists.")
-    sqlite_engine = sa.create_engine(f"sqlite:///{db_path}")
-    try:
-        # So that we can wipe it out
-        pudl.helpers.drop_tables(sqlite_engine, clobber=clobber)
-    except sa.exc.OperationalError:
-        pass
-
-    # And start anew
-    sqlite_engine = sa.create_engine(f"sqlite:///{db_path}")
-    sqlite_meta = sa.MetaData()
-    sqlite_meta.reflect(sqlite_engine)
-
-    # Get the mapping of filenames to table names and fields
-    logger.info(
-        f"Creating a new database schema based on {ferc1_to_sqlite_settings.refyear}."
-    )
-    ferc1_dbf_ds = Ferc1DbfDatastore(datastore)
-    dbc_map = get_dbc_map(ferc1_dbf_ds, ferc1_to_sqlite_settings.refyear)
-    define_sqlite_db(
-        sqlite_engine=sqlite_engine,
-        sqlite_meta=sqlite_meta,
-        dbc_map=dbc_map,
-        ferc1_dbf_ds=ferc1_dbf_ds,
-        ferc1_to_sqlite_settings=ferc1_to_sqlite_settings,
-    )
-
-    for table in DBF_TABLES_FILENAMES.keys():
-        logger.info(f"Pandas: reading {table} into a DataFrame.")
-        new_df = get_raw_df(
-            ferc1_dbf_ds, table, dbc_map, years=ferc1_to_sqlite_settings.years
-        )
-        # Because this table has no year in it, there would be multiple
-        # definitions of respondents if we didn't drop duplicates.
-        if table == "f1_respondent_id":
-            new_df = new_df.drop_duplicates(subset="respondent_id", keep="last")
-        n_recs = len(new_df)
-        logger.debug(f"    {table}: N = {n_recs}")
-        # Only try and load the table if there are some actual records:
-        if n_recs <= 0:
-            continue
-
-        # Write the records out to the SQLite database, and make sure that
-        # the inferred data types are being enforced during loading.
-        # if_exists='append' is being used because we defined the tables
-        # above, but left them empty. Becaue the DB is reset at the beginning
-        # of the function, this shouldn't ever result in duplicate records.
-        coltypes = {col.name: col.type for col in sqlite_meta.tables[table].c}
-        logger.info(f"SQLite: loading {n_recs} rows into {table}.")
-        new_df.to_sql(
-            table,
-            sqlite_engine,
-            if_exists="append",
-            chunksize=100000,
-            dtype=coltypes,
-            index=False,
-        )
+    # TODO(rousik): this thin wrapper seems to be somewhat quirky. Maybe there's a way to make the integration
```

Comment:
Dagster seems to be mostly organized around functions, so I don't know if there's a way to avoid having this class-to-function wrapper. The way the FERC DBF to SQLite extraction happens in the DAG is also a bit different -- with the entire extraction showing up as a standalone job that runs before and separate from all of the downstream operations in the main PUDL ETL. If anyone has thoughts on how to organize this appropriately I suspect it's @bendnorman.

Reply:
It might not be easy to eliminate all of boiler-plate. It seems a bit unwieldy, but it's a minor issue. I can remove this todo and leave as-is.